src/main/java/ymfc/parser/Parser.java

```diff
@@ -136,16 +137,7 @@ private static AddRecipeCommand getAddRecipeCommand(String args) throws InvalidA
                 throw new InvalidArgumentException("Invalid time: " + timeTakenString);
             }
```

Comment:
Do we have to test whether or not `timeTakenString == null`? If yes, I think we should raise an `InvalidArgumentException` too when it's `null` too; otherwise we can just remove the check and put an `assert` instead.

Reply:
I believe no, as time taken is an optional attribute. If the user adds a recipe without the time taken, it should not throw an exception.

Reply:
If it is null, it should not throw an exception as timeTaken is an optional attribute. I think nothing should happen if time taken is not provided as a parameter by the user.

Reply:
Ah so it's nullable, my bad. Excellent work! 😄